pkg/util/docker/dockerfile/dockerfile.go

```diff
@@ -62,3 +62,17 @@ func InsertInstructions(node *parser.Node, pos int, instructions string) error {
 	node.Children = append(node.Children[:pos], append(newChild.Children, node.Children[pos:]...)...)
 	return nil
 }
+
+// NextValues returns a slice of values from the next nodes following node. This
+// roughly translates to the arguments to the Docker builder instruction
+// represented by node.
+func NextValues(node *parser.Node) []string {
```

Comment:
This is a helper for manipulating the AST. Calling it `Arguments` is promising something we cannot fulfill, at least not generically.
For most commands, collecting the values of the linked list formed by the `node.Next` pointers gives the "arguments" of the given command, therefore the usefulness of this helper. It's not true for ONBUILD, for instance. It's also less useful for instructions that take pairs of arguments like ENV.  You have to know what you're feeding the function with.

IMHO, what we can do is to have helpers to work with specific commands. E.g., given a Dockerfile's root node, return:
1. a list of all exposed ports;
2. the image of every from instruction.

WDYT?